test/e2e/suite/conformance/certificates/vault/vault.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vault
+
+import (
+	"path"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/jetstack/cert-manager/test/e2e/framework"
+	"github.com/jetstack/cert-manager/test/e2e/framework/addon/tiller"
+	vault "github.com/jetstack/cert-manager/test/e2e/framework/addon/vault"
+	"github.com/jetstack/cert-manager/test/e2e/suite/conformance/certificates"
+)
+
+var _ = framework.ConformanceDescribe("Certificates", func() {
+	provisioner := new(vaultProvisioner)
+
+	(&certificates.Suite{
+		Name:             "Vault",
+		CreateIssuerFunc: provisioner.create,
+		DeleteIssuerFunc: provisioner.delete,
+	}).Define()
```

Comment:
Are we going to run tests against Vault configured with different authentication types? This seems like an easy way to ensure we have good coverage here 😄

Even if you don't implement additional auth types in this PR, can you update the name of the suite invocation to indicate which auth mechanism is being used, so it's clear when reading test results?

Reply:
Sounds good to me